File: Hackerrank/Balanced Brackets/Balanced Brackets.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the isBalanced function below.
def isBalanced(s):
    # creating a stack
    stack=[]
    
    # iterating for each character in string
    for i in s:
        # if there is open bracket of any type push into stack
        if i=='{' or i=='[' or i=='(':
            stack.append(i)
            
        else:
            # using try and except if element at first index is closed bracket than position -1 is not accessible and return NO in except block
            try:
                # if last element in stack and current element is a matched pair remove the last element from stack
                if stack[-1]=='(' and i==')':
                    stack.pop()
                    
                # if last element in stack and current element is a matched pair remove the last element from stack
                elif stack[-1]=='{' and i=='}':
                    stack.pop()
                    
                # if last element in stack and current element is a matched pair remove the last element from stack
                elif stack[-1]=='[' and i==']':
                    stack.pop()
                
                # otherwise return NO for not balanced
                else:
                    return 'NO'
            except:
                return 'NO'
            
    # if stack is not empty than pairs are not matched. Hence, return NO otherwise YES
    if len(stack)!=0:
        return 'NO'
    else:
        return 'YES'

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        s = input()

        result = isBalanced(s)

        fptr.write(result + '\n')

    fptr.close()

File: Hackerrank/Balanced Brackets/BalancedBrackets.java

static String isBalanced(String s) {
    //create a Stack for storing characters
        Stack <Character> st=new Stack<>();
        boolean isbal=true;
        for(int i=0;i<s.length();i++){
            char ch=s.charAt(i);
            switch(ch){
                //if any of these case is true than push in stack
                case '(':
                case'{':
                case '[':{
                    st.push(ch);
                    continue;
                }
                //if this case is true than break
                case ')':{
                if(st.isEmpty()||st.peek()!='(')
                {
                    isbal=false;
                    break;
                }
                else st.pop();
break;
                }
                  case '}':{
                if(st.isEmpty()||st.peek()!='{')
                {
                    isbal=false;
                    break;
                }
                else st.pop();
                break;
                }
                  case ']':{
                if(st.isEmpty()||st.peek()!='[')
                {
                    isbal=false;
                    break;
                }
                else st.pop();
                break;
                }
            }
        }
        // if stack is not empty that means it is not balance;
         if(!st.isEmpty()){
                    isbal=false;
                }
return (isbal)?"YES":"NO";

    }
